modules/data.land/R/soilgrids_soc_extraction.R

```diff
@@ -101,12 +102,9 @@ soilgrids_soilC_extract <- function (site_info, outdir=NULL, verbose=TRUE) {
     dep.ind <- which(grepl(depths[dep], name_tag[, 1]))
     ocdquant[dep, ] <- ocd_real[dep.ind] %>% unlist
   }
-  na.ind <- which(is.na(ocdquant[1, 1:length(site_info$site_id)]))
-  internal_site_info <- data.frame(site_info$site_id[-na.ind], 
-                                   site_info$site_name[-na.ind], 
-                                   site_info$lat[-na.ind],
-                                   site_info$lon[-na.ind]) %>% `colnames<-`(names(site_info))
-  
+  na_sites <- is.na(ocdquant[1, seq_len(nrow(internal_site_info))])
+  internal_site_info <- internal_site_info[!na_sites, ]
```

Comment:
Here's the bugfix: `x[-which(all_FALSEs), ]` returns an empy df, but `x[!all_FALSEs, ]` returns x as desired.